--- a/Makefile	2022-06-07 15:05:10.039804242 +0100
+++ b/Makefile	2022-06-07 15:06:42.146835573 +0100
@@ -281,7 +281,7 @@
 SDL_LIBS:=$(shell $(PKG_CONFIG) $(SDL_PCNAME) --libs)
 SDL_LIBS+=$(call linktest,SDL2_image,-lSDL2_image,$(SDL_LIBS))
 SDL_LIBS+=$(call linktest,SDL2_ttf,-lSDL2_ttf,$(SDL_LIBS))
-SDL_LIBS+=$(call linktest,SDL2_gfx,-lSDL2_gfx,$(SDL_LIBS))
+SDL_LIBS+=$(shell $(PKG_CONFIG) SDL2_gfx --libs)
 SDL_LIBS+=$(call linktest,zlib,-lz,)
 SDL_LIBS+=$(call linktest,libpng,$(PNG),)
 
@@ -296,6 +296,7 @@
 SDL_LIBS+=$(SDL_MIXER_LIB) $(SDL2_PANGO_LIB)
 
 SDL_CFLAGS:=$(shell $(PKG_CONFIG) $(SDL_PCNAME) --cflags)
+SDL_CFLAGS+=$(shell $(PKG_CONFIG) SDL2_gfx --cflags)
 
 
 # New one: -lrsvg-2 -lcairo
