From ed66f579acc4eb7a94dcff80dd6f4fb19abf0581 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 11 Feb 2017 14:00:53 -0800
Subject: [PATCH 1/3] build: don't install bash completion in /etc unless
 system-wide

---
 Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 730ed91..51cc780 100644
--- a/Makefile
+++ b/Makefile
@@ -148,7 +148,11 @@ MAN_PREFIX:=$(DESTDIR)$(PREFIX)/share/man
 DEVMAN_PREFIX:=$(DESTDIR)$(PREFIX)/share/man
 
 # BASH tab-completion file:
-COMPLETIONDIR:=$(DESTDIR)/etc/bash_completion.d
+ifeq ($(PREFIX),/usr)
+  COMPLETIONDIR:=$(DESTDIR)/etc/bash_completion.d
+else
+  COMPLETIONDIR:=$(DESTDIR)$(PREFIX)/etc/bash_completion.d
+endif
 
 # 'System-wide' Config file:
 ifeq ($(PREFIX),/usr)
-- 
2.9.3

