From e799a32454298ab339af4d23e938d88f1f1d100b Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 11 Feb 2017 13:43:22 -0800
Subject: [PATCH 3/3] Install desktop file and icons

---
 Makefile | 118 ++++++++++++++++++++++++++++++++++-----------------------------
 1 file changed, 63 insertions(+), 55 deletions(-)

diff --git a/Makefile b/Makefile
index 0b9eb86..bd62925 100644
--- a/Makefile
+++ b/Makefile
@@ -172,6 +172,13 @@ GNOME_PREFIX:=$(shell gnome-config --prefix 2> /dev/null)
 KDE_PREFIX:=$(shell kde-config --install apps --expandvars 2> /dev/null)
 KDE_ICON_PREFIX:=$(shell kde-config --install icon --expandvars 2> /dev/null)
 
+ifneq ($(KDE_ICON_PREFIX),)
+  XDG_ICON_PREFIX:=$(KDE_ICON_PREFIX)
+else
+  XDG_ICON_PREFIX:=$(DESTDIR)$(PREFIX)/share/icons
+endif
+
+
 # Maemo flag
 MAEMOFLAG:=
 
@@ -439,7 +446,7 @@ trans:
 windows_ARCH_INSTALL:=
 osx_ARCH_INSTALL:=
 beos_ARCH_INSTALL:=install-haiku
-linux_ARCH_INSTALL:=install-gnome install-kde install-kde-icons
+linux_ARCH_INSTALL:=install-gnome install-kde install-xdg-icons
 ARCH_INSTALL:=$($(OS)_ARCH_INSTALL)
 
 # "make install" installs all of the various parts
@@ -581,17 +588,15 @@ uninstall:	uninstall-i18n
 	fi
 	-rm $(ICON_PREFIX)/tuxpaint.png
 	-rm $(X11_ICON_PREFIX)/tuxpaint.xpm
-	-if [ "x$(KDE_ICON_PREFIX)" != "x" ]; then \
-	  rm $(KDE_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
-	  rm $(KDE_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
-	fi
+	-rm $(XDG_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
+	-rm $(XDG_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
 	-rm $(BIN_PREFIX)/tuxpaint
 	-rm $(BIN_PREFIX)/tuxpaint-import
 	-rm -r $(DATA_PREFIX)
@@ -719,18 +724,23 @@ echo-install-example-templates:
 install-example-templates: echo-install-example-templates install-example-template-dirs $(INSTALLED_TEMPLATES)
 
 
-# Install a launcher icon in the Gnome menu
+# Install a launcher icon in the Gnome/xdg menu
 .PHONY: install-gnome
 install-gnome:
 	@echo
-	@echo "...Installing launcher icon into GNOME..."
 	@if [ "x$(GNOME_PREFIX)" != "x" ]; then \
+	 echo "...Installing launcher icon into GNOME..."; \
 	 install -d $(DESTDIR)$(GNOME_PREFIX)/share/pixmaps; \
 	 cp data/images/icon.png $(DESTDIR)/$(GNOME_PREFIX)/share/pixmaps/tuxpaint.png; \
 	 chmod 644 $(DESTDIR)$(GNOME_PREFIX)/share/pixmaps/tuxpaint.png; \
 	 install -d $(DESTDIR)$(GNOME_PREFIX)/share/applications; \
 	 cp src/tuxpaint.desktop $(DESTDIR)$(GNOME_PREFIX)/share/applications/; \
 	 chmod 644 $(DESTDIR)$(GNOME_PREFIX)/share/applications/tuxpaint.desktop; \
+	else \
+	 echo "...Installing launcher into XDG..."; \
+	 install -d $(DESTDIR)$(PREFIX)/share/applications; \
+	 cp src/tuxpaint.desktop $(DESTDIR)$(PREFIX)/share/applications/; \
+	 chmod 644 $(DESTDIR)$(PREFIX)/share/applications/tuxpaint.desktop; \
 	fi
 
 
@@ -770,47 +780,45 @@ install-kde:
 	  chmod 644 $(DESTDIR)$(KDE_PREFIX)/Graphics/tuxpaint.desktop; \
 	fi
 
-.PHONY: install-kde-icons
-install-kde-icons:
-	@echo "...Installing launcher icon graphics into KDE..."
-	@if [ "x$(KDE_ICON_PREFIX)" != "x" ]; then \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/scalable/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/192x192/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/128x128/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/96x96/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/64x64/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/48x48/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/32x32/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/22x22/apps/; \
-	  install -d $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/16x16/apps/; \
-	  cp data/images/tuxpaint-icon.svg \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
-	  cp data/images/icon192x192.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
-	  cp data/images/icon128x128.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
-	  cp data/images/icon96x96.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
-	  cp data/images/icon64x64.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
-	  cp data/images/icon48x48.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
-	  cp data/images/icon32x32.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
-	  cp data/images/icon22x22.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
-	  cp data/images/icon16x16.png \
-		$(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
-          chmod 644 $(DESTDIR)$(KDE_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
-	fi
+.PHONY: install-xdg-icons
+install-xdg-icons:
+	@echo "...Installing launcher icon graphics into XDG..."
+	install -d $(XDG_ICON_PREFIX)/hicolor/scalable/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/192x192/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/128x128/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/96x96/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/64x64/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/48x48/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/32x32/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/22x22/apps/; \
+	install -d $(XDG_ICON_PREFIX)/hicolor/16x16/apps/; \
+	cp data/images/tuxpaint-icon.svg \
+		$(XDG_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/scalable/apps/tuxpaint.svg; \
+	cp data/images/icon192x192.png \
+		$(XDG_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/192x192/apps/tuxpaint.png; \
+	cp data/images/icon128x128.png \
+		$(XDG_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/128x128/apps/tuxpaint.png; \
+	cp data/images/icon96x96.png \
+		$(XDG_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/96x96/apps/tuxpaint.png; \
+	cp data/images/icon64x64.png \
+		$(XDG_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/64x64/apps/tuxpaint.png; \
+	cp data/images/icon48x48.png \
+		$(XDG_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/48x48/apps/tuxpaint.png; \
+	cp data/images/icon32x32.png \
+		$(XDG_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/32x32/apps/tuxpaint.png; \
+	cp data/images/icon22x22.png \
+		$(XDG_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/22x22/apps/tuxpaint.png; \
+	cp data/images/icon16x16.png \
+		$(XDG_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
+        chmod 644 $(XDG_ICON_PREFIX)/hicolor/16x16/apps/tuxpaint.png; \
 
 
 # Install the PNG icon (for GNOME, KDE, etc.)
-- 
2.9.3

