From 72a8bef0cd1b8057e0f71f988bcb6796fecbf816 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 15 Aug 2020 07:29:48 +0100
Subject: [PATCH 2/3] Set flatpak /app profile

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index ec8007b8..7fe1cf1e 100644
--- a/Makefile
+++ b/Makefile
@@ -104,7 +104,7 @@ ARCH_CFLAGS:=$($(OS)_ARCH_CFLAGS)
 windows_ARCH_LDFLAGS:=
 osx_ARCH_LDFLAGS:=-L/opt/local/lib
 beos_ARCH_LDFLAGS:=
-linux_ARCH_LDFLAGS:=
+linux_ARCH_LDFLAGS:=-L/app/lib
 ARCH_LDFLAGS:=$($(OS)_ARCH_LDFLAGS)
 LDFLAGS:=$(ARCH_LDFLAGS)
 
@@ -131,7 +131,7 @@ ARCH_HEADERS:=$($(OS)_ARCH_HEADERS)
 # For macOS, the prefix is relative to DESTDIR.
 windows_PREFIX:=/usr/local
 osx_PREFIX:=Resources
-linux_PREFIX:=/usr/local
+linux_PREFIX:=/app
 PREFIX:=$($(OS)_PREFIX)
 
 # Root directory to place files when creating packages.
-- 
2.26.2

